mobile-app/app/components/BottomSheetModal.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactElement, useCallback, useRef, useEffect } from 'react'
+import { tailwind } from '@tailwind'
+import { TouchableOpacity, View, Platform } from 'react-native'
+import { BottomSheetModal as Modal, BottomSheetModalProps, useBottomSheetModal } from '@gorhom/bottom-sheet'
+import { useThemeContext } from '@contexts/ThemeProvider'
+import { ThemedProps } from './themed'
+
+type Props = ThemedProps & BottomSheetModalProps & {
+  triggerComponent: ReactElement
+  children: ReactElement
+}
+
+export const BottomSheetModal = (props: Props): JSX.Element => {
+  const bottomSheetModalRef = useRef<Modal>(null)
+  const { dismiss } = useBottomSheetModal()
+  const { isLight } = useThemeContext()
+  const {
+    name,
+    style,
+    children,
+    triggerComponent,
+    snapPoints = ['100%', '50%'],
+    light = tailwind('bg-gray-100 text-black'),
+    dark = tailwind('bg-gray-600 text-white text-opacity-90')
+  } = props
+
+  const openModal = useCallback(() => {
+    bottomSheetModalRef.current?.present()
+  }, [])
+
+  useEffect(() => {
+    return () => {
+      if (Platform.OS !== 'web') {
+        dismiss(name)
+      }
+    }
+  }, [])
+
+  if (Platform.OS === 'web') {
+    return <></>
+  }
+
+  return (
+    <View>
+      <TouchableOpacity onPress={openModal}>
+        {triggerComponent}
+      </TouchableOpacity>
+      <Modal
+        name={name}
+        ref={bottomSheetModalRef}
+        style={style}
+        snapPoints={snapPoints}
+        stackBehavior='replace'
+        backgroundStyle={[isLight ? light : dark]}
+      >
+        {children}
```

Comment:
Also from figma the backdrop has a dimmed color, possible to add that using [this](https://gorhom.github.io/react-native-bottom-sheet/props#backdropcomponent)?

Reply:
In figma its grey-50, which matches the background colour in light mode and in order to distinguish between modal and the main page we are using grey-100